### PR TITLE
Add new "intent" prefetch strategy

### PR DESCRIPTION
**Pull request by pleek91**
# Description
The new "intent" prefetch strategy triggers prefetching when a router link emits a [focusin](https://developer.mozilla.org/en-US/docs/Web/API/Element/focusin_event) or [mouseover](https://developer.mozilla.org/en-US/docs/Web/API/Element/mouseover_event) event. This allows for prefetching to happen right before a user navigates to the route.

### CODE REVIEW

File: src/components/routerLink.browser.spec.ts [hunk 1]
```diff
@@ -329,7 +329,7 @@ test('isMatch correctly matches parent when sibling has the same url', async ()
     parent: parentRoute,
     name: 'child',
     path: '/child',
-    component: () => h(routerLink, { to: (resolve) => resolve('parent') }, 'parent'),
+    component: () => h(routerLink, { to: (resolve) => resolve('parent') }, () => 'parent'),
```
Comment (pleek91) [inline src/components/routerLink.browser.spec.ts:332]: Unrelated fix for a vue warning when running tests.

File: src/compositions/useVisibilityObserver.ts [hunk 2]
```diff
@@ -25,6 +25,12 @@ export function useVisibilityObserver(element: Ref<Element | undefined>): UseVis
     }
   }, { immediate: true })
 
+  onUnmounted(() => {
+    if (element.value) {
+      observer.unobserve(element.value)
+    }
+  })
+
```
Comment (pleek91) [inline src/compositions/useVisibilityObserver.ts:33]: unrelated fix to make sure elements are no longer observed after they are unmounted.

Comment (pleek91) [reply src/compositions/useVisibilityObserver.ts:33]: Actually I'm going to make a separate PR for this so it will show up as a fix in the change log.

**Review by stackoverfloweth — APPROVED** [review]
mad kudos dude, this is a complicated feature implemented with +136-3 🏆
